configure.ac

```diff
@@ -200,25 +200,22 @@ AC_ARG_WITH([cuse],
 )
 
 if test "$with_cuse" != "no"; then
-    LIBFUSE_CFLAGS=$(pkg-config fuse --cflags 2>/dev/null)
-    if test $? -ne 0; then
-        if test "$with_cuse" = "yes"; then
-            AC_MSG_ERROR("Is fuse-devel installed? -- could not get cflags for libfuse")
-        else
-            with_cuse=no
-        fi
-    else
-        with_cuse=yes
-    fi
+	if pkg-config fuse3; then
+		LIBFUSE_CFLAGS=$(pkg-config fuse3 --cflags)
+		LIBFUSE_LIBS=$(pkg-config fuse3 --libs)
+		with_cuse=yes
+	elif pkg-config fuse; then
+		LIBFUSE_CFLAGS="$(pkg-config fuse --cflags) -DWITH_FUSE2"
+		LIBFUSE_LIBS=$(pkg-config fuse --libs)
+		with_cuse=yes
+	else
+		AC_MSG_ERROR("Is fuse3-devel or fuse-devel installed? -- could not found libfuse3 or libfuse")
+	fi
 fi
```

Comment:
When none of them is installed, then this doesn't work anymore:

```
./configure --prefix=/usr
[...]
checking for clock_gettime in -lc... yes
checking for whether to build with CUSE interface... configure: error: "Is fuse3-devel or fuse-devel installed? -- could not found libfuse3 or libfuse"
```

Reply:
actually this is the expected behaviour, when user specified `--with-cuse` but neither fuse3 nor fuse2 is installed, `./configure` should fail and notify about it.

or should it modify `with_cuse` to `no` and then continue? maybe not, because the product it build does not contains `swtpm_cuse`, which is a mismatching to `--with-cuse` flag.

Reply:
I did not `specify --with-cuse` and in this case it previously was building with CUSE when it detected fuse , did not build CUSE otherwise but was throwing an error if it was request with `--with-cuse` but fuse could not be found. I just want the old behavior. My suggestion for when to error out with AC_MSG_ERROR should fix it.